File: Daily Challenge/C++/Integers - Strictly Increasing Or Not 17.8.2025.cpp
#include <bits/stdc++.h>
using namespace std;

struct Pair {
    char ch;
    int num;
};

bool cmp(Pair a, Pair b) {
    return a.ch < b.ch;
}

int main() {
    int N;
    cin >> N;
    vector<Pair> arr(N);
    for (int i = 0; i < N; i++) {
        string s;
        cin >> s;
        arr[i].num = stoi(s);
        arr[i].ch = tolower(s.back());
    }
    sort(arr.begin(), arr.end(), cmp);
    bool ok = true;
    for (int i = 0; i < N-1; i++) {
        if (arr[i].num >= arr[i+1].num) {
            ok = false;
            break;
        }
    }
    cout << (ok ? "Yes" : "No") << endl;
}

File: Daily Challenge/Java/Integers - Strictly Increasing Or Not 17.8.2025.java
import java.util.*;

class Main {
    static class Pair {
        char ch;
        int num;
        Pair(char c, int n) { ch = c; num = n; }
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int N = Integer.parseInt(sc.nextLine());
        List<Pair> list = new ArrayList<>();

        for (int i = 0; i < N; i++) {
            String s = sc.nextLine().trim();
            int num = Integer.parseInt(s.substring(0, s.length()-1));
            char ch = Character.toLowerCase(s.charAt(s.length()-1));
            list.add(new Pair(ch, num));
        }

        list.sort(Comparator.comparing(p -> p.ch));
        boolean ok = true;
        for (int i = 0; i < N-1; i++) {
            if (list.get(i).num >= list.get(i+1).num) {
                ok = false;
                break;
            }
        }
        System.out.println(ok ? "Yes" : "No");
    }
}